### ISSUE TITLE
 Can DPM-Solver support self-conditioning?

### ISSUE THREAD

**Issue by summmeer** [LuChengTHU/dpm-solver#48]
Hi,
Thank you for your contribution and I've successfully adapted dpm-solver to text generation efficiently. Now I'm curious about: 
Can current dpm-solver support self-conditioning (https://arxiv.org/abs/2208.04202), since this technique is widely used in many text generation application. 

Any response is appreciated!

**Comment by LuChengTHU** [comment]
I believe it can also works, all we need to do is to modify the `model_wrapper`

**Comment by summmeer** [comment]
Yes! We successfully integrate dmp-solver into our model in our project [DoT](https://github.com/HKUNLP/diffusion-of-thoughts). Thanks a lot!
